### PR TITLE
Add information about test integrations automation

### PR DESCRIPTION
**Pull request by mrodm**
This PR adds a new section in the README to include the information about the `test integrations` automation.

This PR also merges the two Development sections that were available in the README.

Rendered README can be checked [here](https://github.com/mrodm/elastic-package/blob/update_readme_test_integrations/README.md#development).

Follow-up https://github.com/elastic/elastic-package/pull/1277

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -512,21 +562,6 @@ The following settings are available per profile:
   Elasticsearch in stacks managed by elastic-package. It is recommended to use
   an absolute path, out of the `.elastic-package` directory.
 
-## Development
```
Comment (mrodm) [inline README.md:515]: This section was duplicated. Merged into the other Development section in line 39

File: README.md [hunk 2]
```diff
@@ -38,7 +38,26 @@ elastic-package help
...
+
+`make check` - one-liner, used by CI to verify if source code is ready to be pushed to the repository
+
+`make install` - build the tool source and move binary to `$GOBIN`
+
+`make gomod` - ensure go.mod and go.sum are up to date
+
+`make update` - update README.md file
+
+`make licenser` - add the Elastic license header in the source code
```
Comment (mrodm) [inline README.md:58]: Added a few more targets, and change `make vendor` target by the `make gomod` one.

**Review by jlind23 — APPROVED** [review]

File: tools/readme/readme.md.tmpl [hunk 3]
```diff
@@ -61,6 +80,37 @@ git rm --cached -r .
...
+   go mod edit -replace github.com/elastic/elastic-package=github.com/<your_github_user>/elastic-package@1131866bcff98c29e2c84bcc1c772fff4307aaca
+   go mod tidy
+   ```
+4. Push these changes into a branch and create a Pull Request
+    - Creating this PR would automatically trigger a new Jenkins pipeline.
+
+
+This process can also be done automatically directly from your Pull Request by adding a comment `test integrations`. Example:
+- Comment: https://github.com/elastic/elastic-package/pull/1335#issuecomment-1619721861
+- Pull Request created in integrations repository: https://github.com/elastic/integrations/pull/6756
```
Comment (jsoriano) [inline tools/readme/readme.md.tmpl:108]: Perhaps we should mention this first, before the detailed steps.

**Review by jsoriano — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**